src/Surface.zig

```diff
@@ -2879,6 +2819,64 @@ pub fn mouseButtonCallback(
         }
     }
 
+    // Set the clipboard if configured with copy_on_select
+    if (button == .left and
+        action == .release and
+        self.config.copy_on_select != .false)
+    set_clipboard: {
+        self.renderer_state.mutex.lock();
+        defer self.renderer_state.mutex.unlock();
+
+        // Only update clipboard if something is actually selected.
+        const selection = self.io.terminal.screen.selection orelse break :set_clipboard;
+
+        const buf = self.io.terminal.screen.selectionString(self.alloc, .{
+            .sel = selection,
+            .trim = self.config.clipboard_trim_trailing_spaces,
+        }) catch |err| {
+            log.err("error reading selection string err={}", .{err});
+            break :set_clipboard;
+        };
+        log.debug("copying {} to clipboards", .{selection});
```

Comment:
Remove this log, especially since it could potentially leak sensitive data into the logs (like passwords or security tokens).